src/commands/importFunction/importFunctionApp.ts

```diff
@@ -0,0 +1,405 @@
+        return hostKeys.functionKeys[funcAppKeyName];
+    }
+    ext.outputChannel.appendLine(localize("importFunctionApp", `Can't find existing function app host key ${funcAppKeyName}`));
+    return await createFuncHostKey(baseUrl, funcAppKeyName, credentials);
+}
+
+// Create function hostkey for our apim service
+async function createFuncHostKey(baseUrl: string, funcKeyName: string, credentials: ServiceClientCredentials): Promise<string> {
+    ext.outputChannel.appendLine(localize("importFunctionApp", `Create new function app host key ${funcKeyName}`));
+    // create new webresource
+    const options = new WebResource();
+    options.method = "PUT";
+    options.url = `${baseUrl}/host/default/functionkeys/${funcKeyName}?api-version=2018-11-01`;
+    options.headers = {
+        ['User-Agent']: appendExtensionUserAgent()
+    };
+    await signRequest(options, credentials);
+    // send request to get new function hostkey
+    const funKeyRequest = <nRequest>options;
+    funKeyRequest.body = {
+        properties: {}
+    };
+    funKeyRequest.json = true;
+    const response = await sendRequest(funKeyRequest);
+    if (response.properties.value) {
+        return String(response.properties.value);
+    } else {
+        throw new Error("Unexpected null value when generating function host key");
+    }
+}
+
+// Get function hostkey
+async function getFuncHostKeys(baseUrl: string, credentials: ServiceClientCredentials): Promise<IFunctionKeys> {
+    const options = new WebResource();
+    options.method = "POST";
+    options.url = `${baseUrl}/host/default/listkeys?api-version=${Constants.functionAppApiVersion}`;
+    options.headers = {
+        ['User-Agent']: appendExtensionUserAgent()
+    };
+    await signRequest(options, credentials);
+    // tslint:disable-next-line: await-promise
+    const funcKeys = await request(options).promise();
+    // tslint:disable-next-line: no-unsafe-any
+    return JSON.parse(funcKeys);
+}
+
+// create new operation for each function and method
+function getNewOperation(apiId: string, method: string, name: string, route: string, displayName: string): OperationContract {
+    const operationId = apiUtils.displayNameToIdentifier(name);
+    const cleanUrl = Utils.parseUrlTemplate(route);
+    return { id: `${apiId}/operations/${operationId}`, name: operationId, displayName: displayName, method: method, urlTemplate: cleanUrl.urlTemplate, description: "", templateParameters: cleanUrl.parameters };
+}
+
+function getFunctionAppBase(invokeUrlTemplate: string, route: string): string {
+    const uri = invokeUrlTemplate.substr(0, invokeUrlTemplate.length - route.length);
+    return uri.endsWith("/") ? uri.substr(0, uri.length - 1) : uri;
+}
+
+async function createApiFromFunctionApp(apiId: string, funcApp: Site, apiName: string): Promise<ApiContract> {
+    ext.outputChannel.appendLine(localize("importFunctionApp", `Create Api for the function app ${funcApp.name}...`));
+    if (funcApp.name) {
```

Comment:
no need for this check

Reply:
since funcApp.name can be (string | undefined), previously want to avoid no-null-assertion error, anyway, I will remove it